src/runtime_lib/infra_gapbs/intersections.h

```diff
@@ -0,0 +1,336 @@
+// Copyright (c) 2015, The Regents of the University of California (Regents)
+// See LICENSE.txt for license details
+
+#include <algorithm>
+#include <cinttypes>
+#include <iostream>
+#include <vector>
+#include <fstream>
+#include "benchmark.h"
+#include "builder.h"
+#include "command_line.h"
+#include "graph.h"
+#include "pvector.h"
+#include "bitmap.h"
+#include "timer.h"
+
+
+using namespace std;
+
+
+/*
+Runtime library for various set intersection methods
+*/
+
+
+
+size_t MAX_LIMIT = 61578415;
+
+
+//TODO: there might be some issues with the allocating space (for now I am just allocating big enough space)
+size_t intersectSortedNodeSetBitset(NodeID* A, NodeID* B, size_t totalA, size_t totalB) {
+  
+  size_t total = 0;
+
+  //we only want to build bitmap for bigger array
+  if (totalA > totalB) {
+
+    Bitmap a_array(MAX_LIMIT);
```

Comment:
Here we should just assume a bitarrray will be passed in as an argument. It is not going to be worthwhile to construct a new bitarray just for one intersection.